docs/tasks.md

```diff
@@ -581,14 +582,22 @@ The `description` field is an optional field that allows you to add an informati
 
 ### Using variable substitution
 
+Tekton provides variables to inject values into the contents of certain fields.
+The values you can inject come from a range of sources including other fields
+in the Task or Pipeline, context-sensitive information that Tekton provides,
+and runtime information received from a TaskRun or PipelineRun.
+
+The mechanism of variable substitution is quite simple - string replacement is
+performed by the Tekton Controller when a TaskRun or PipelineRun is executed.
+
```

Comment:
it would be helpful to have this in [pipelines docs](https://github.com/tektoncd/community/pull/293) because it's also just referencing the `variables.md` as well, what do you think?

Reply:
Yeah, that makes sense, I'll update this PR to add details to the pipelines doc too.